cpmpy/solvers/cpo.py

```diff
@@ -0,0 +1,491 @@
+import shutil
+
+from .solver_interface import SolverInterface, SolverStatus, ExitStatus
+from .. import DirectConstraint
+from ..expressions.core import Expression, Comparison, Operator, BoolVal
+from ..expressions.globalconstraints import GlobalConstraint
+from ..expressions.globalfunctions import GlobalFunction
+from ..expressions.variables import _BoolVarImpl, NegBoolView, _IntVarImpl, _NumVarImpl
+from ..expressions.utils import is_num, is_any_list, eval_comparison, argval, argvals
+from ..transformations.get_variables import get_variables
+from ..transformations.normalize import toplevel_list
+from ..transformations.decompose_global import decompose_in_tree
+
+"""
+    Interface to CP Optimizers API
+
+    <some information on the solver>
```

Comment:
yes : )
From their own docs:
CP Optimizer, also a feature of IBM ILOG Optimization Studio, is a software library of constraint programming tools supporting constraint propagation, domain reduction, and highly optimized solution search.